tests/test_dataclasses.py

```diff
@@ -1647,6 +1649,18 @@ class B(A):
     assert B(1, y=2, z=3) == B(x=1, y=2, z=3)
 
 
+@pytest.mark.parametrize('field_constructor', [pydantic.dataclasses.Field, dataclasses.field])
+def test_repr_false(field_constructor: Callable):
+    @pydantic.dataclasses.dataclass
+    class A:
+        visible_field: str
+        hidden_field: str = field_constructor(repr=False)
+
+    instance = A(visible_field='this_should_be_included', hidden_field='this_should_not_be_included')
+    assert "visible_field='this_should_be_included'" in repr(instance)
+    assert "hidden_field='this_should_not_be_included'" not in repr(instance)
+
+
```

Comment:
These changes look fantastic 👍

Reply:
Thanks for the feedback! 

re: https://github.com/pydantic/pydantic/issues/8454, are you already working on it, or should I take a stab at it?

Reply:
@tigeryy2, I haven't started yet - feel free to take a stab! Thanks : ).